validation/validation_test.go

```diff
@@ -17,54 +22,90 @@ var (
 	runtime = "runc"
 )
 
+type validation struct {
+	test        func(string, string, *rspec.Spec) error
+	description string
+}
+
 func init() {
 	runtime = os.Getenv("RUNTIME")
 }
 
-func runtimeValidate(runtime string, g *generate.Generator) error {
+func TestValidateRuntimeInside(t *testing.T) {
+	g, err := getDefaultGenerator()
+	if err != nil {
+		t.Errorf("%s failed validation: %v", runtime, err)
+	}
+	g.SetProcessArgs([]string{"/runtimetest"})
+
+	if err := runtimeInsideValidate(runtime, g); err != nil {
+		t.Errorf("%s failed validation: %v", runtime, err)
+	}
+}
+
+func TestValidateRuntimeOutside(t *testing.T) {
+	g, err := getDefaultGenerator()
+	if err != nil {
+		t.Errorf("%s failed validation: %v", runtime, err)
+	}
+	g.SetProcessArgs([]string{"sleep", "50"})
```

Comment:
Why are we setting this at all, since `sleep` may not exist inside the container?  [`process` is optional][1] (since opencontainers/runtime-spec#701), so I'd rather just unset `process` completely and call `create` (but never `start`) during these outer tests.

[1]: https://github.com/opencontainers/runtime-spec/blob/b5017e22ec9592d9e4935812fdce8f9a6b71a365/config.md#process